solver/llbsolver/file/backend.go

```diff
@@ -67,6 +67,40 @@ func mkdir(d string, action *pb.FileActionMkDir, user *copy.User, idmap *idtools
 	return nil
 }
 
+func symlink(d string, action *pb.FileActionSymlink, user *copy.User, idmap *idtools.IdentityMapping) (err error) {
+	defer func() {
+		var osErr *os.PathError
+		if errors.As(err, &osErr) {
+			// remove system root from error path if present
+			osErr.Path = strings.TrimPrefix(osErr.Path, d)
+		}
+	}()
+
+	src, err := fs.RootPath(d, filepath.Join("/", action.Src))
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	ch, err := mapUserToChowner(user, idmap)
+	if err != nil {
+		return err
+	}
+
+	if err := os.Symlink(action.Target, src); err != nil {
+		return errors.WithStack(err)
+	}
+
+	if err := copy.Chown(src, nil, ch); err != nil {
```

Comment:
@profnandaa What's the wcow implications of this. Not sure which of these functions work correctly there (or if they work on symlink or the target).

Reply:
Let me check and revert.

Reply:
### My preliminary findings

When I run the test on Windows (commenting out `// requiresLinux(t)`), I get this:
```
=== NAME  TestIntegration/TestFileOpSymlink/worker=containerd
    client_test.go:2482: 
        	Error Trace:	C:/dev/buildkit/client/client_test.go:2482
        	            				C:/dev/buildkit/util/testutil/integration/run.go:97
        	            				C:/dev/buildkit/util/testutil/integration/run.go:211
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 7777
```

Then the platform time differences:

```

=== NAME  TestIntegration/TestFileOpSymlink/worker=containerd
    client_test.go:2495: 
                Error Trace:    C:/dev/buildkit/client/client_test.go:2495
                                                        C:/dev/buildkit/util/testutil/integration/run.go:97
                                                        C:/dev/buildkit/util/testutil/integration/run.go:211
                Error:          Not equal:
                                expected: time.Date(2024, time.November, 19, 23, 58, 26, 0, time.Local)
                                actual  : time.Date(1969, time.December, 31, 16, 0, 42, 0, time.Local)

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,2 @@
                                -(time.Time) 2024-11-19 23:58:26 -0800 PST
                                +(time.Time) 1969-12-31 16:00:42 -0800 PST
```

Passes when I comment out those:
```diff
diff --git a/client/client_test.go b/client/client_test.go
index 840135094..667c4de01 100644
--- a/client/client_test.go
+++ b/client/client_test.go
@@ -2411,7 +2411,7 @@ func testOCILayoutPlatformSource(t *testing.T, sb integration.Sandbox) {
 }
 
 func testFileOpSymlink(t *testing.T, sb integration.Sandbox) {
-       requiresLinux(t)
+       // requiresLinux(t)
 
        const (
                fileOwner = 7777
@@ -2479,8 +2479,8 @@ func testFileOpSymlink(t *testing.T, sb integration.Sandbox) {
        require.NoError(t, err)

        require.Equal(t, []byte("contents"), dt)
-       require.Equal(t, header.Uid, fileOwner)
-       require.Equal(t, header.Gid, fileGroup)
+       // require.Equal(t, header.Uid, fileOwner)
+       // require.Equal(t, header.Gid, fileGroup)

        entry, ok = m["baz"]
        dt = entry.Data
@@ -2489,10 +2489,10 @@ func testFileOpSymlink(t *testing.T, sb integration.Sandbox) {

        // make sure it was chowned properly
        require.Equal(t, true, ok)
-       require.Equal(t, header.Uid, linkOwner)
-       require.Equal(t, header.Gid, linkGroup)
+       // require.Equal(t, header.Uid, linkOwner)
+       // require.Equal(t, header.Gid, linkGroup)

-       require.Equal(t, header.ModTime, dummyTime)
+       // require.Equal(t, header.ModTime, dummyTime)
 }
```

Reply:
Sorry, with my diff, I didn't mean that you change your tests. Please go ahead and return the `requiresLinux` and leave your previous tests as-is. I was just testing on how it could run on Windows ... we'll still need to figure out permission stuff on Windows, that's a separate issue.

Reply:
I'll revert and open an issue